los_keeper_ros2/src/los_server/los_server.cc

```diff
@@ -1,13 +1,84 @@
 #include "los_keeper_ros2/los_server/los_server.h"
 
 using namespace los_keeper;
+
+std::string LosServer::ProcessMessage(const std::string &raw_string) {
+  std::string result_string;
+  for (char c : raw_string) {
+    result_string.push_back(c);
+    result_string += ' ';
+    std::this_thread::sleep_for(std::chrono::milliseconds(100));
+  }
+  return result_string;
+}
+
+void LosServer::ShortTopicCallback(const std_msgs::msg::String::SharedPtr msg) {
+  auto current_time = now();
+  std::string received_string =
+      msg->data + " (" + std::to_string(current_time.seconds()) + ")";
+  auto processed_string = ProcessMessage(received_string);
+  {
+    std::unique_lock<std::mutex> lock(mutex_list_.short_topic, std::defer_lock);
+    if (lock.try_lock()) {
+      RCLCPP_INFO(this->get_logger(), "Could set short message: %s",
+                  processed_string.c_str());
+      this->wrapper_.SetShortString(processed_string);
+    } else {
+      RCLCPP_INFO(this->get_logger(),
+                  "Unlock failed when setting short message: %s",
+                  processed_string.c_str());
+    }
+  }
+};
+void LosServer::LongTopicCallback(const std_msgs::msg::String::SharedPtr msg) {
+  auto current_time = now();
+  std::string received_string =
+      msg->data + " (" + std::to_string(current_time.seconds()) + ")";
+  auto processed_string = ProcessMessage(received_string);
+  {
+    std::unique_lock<std::mutex> lock(mutex_list_.short_topic, std::defer_lock);
```

Comment:
"mutex_list_.short_topic" should be "mutex_list_.long_topic", right?

Reply:
Ah right! My mistake. I will revise and re push